### DISCUSSION TITLE
Dither Edges of gif. How to make them smooth?

### DISCUSSION THREAD

**Discussion by hafizmabuzar** [dlemstra/Magick.NET#746] [Help]
I get jaggie edges of gif when I create it from png images. How can I make the edges smooth.
The png images have the smooth edges but the output gif has the dither edges.
![sample](https://user-images.githubusercontent.com/21033535/94666228-a4980980-0326-11eb-87c8-f9e5360d6a62.gif)

Any solution will be highly appreciated.
Here is my code:

 using (MagickImageCollection collection = new MagickImageCollection())
            {
                ImageOperations oImageOperations = new ImageOperations();
                int DelayIndex = 10;
                for (double k = 1; k <= MaxFrames; k++)
                {
                    string ImgPath = TempDirPath + "/" + ecover_name + "_" + k.ToString() + ".png";

                    MagickImage img = new MagickImage(oImageOperations.GetStreamedImage(ImgPath));
                    img.AnimationDelay = DelayIndex;
                    img.GifDisposeMethod = GifDisposeMethod.Background;
                    img.AntiAlias = true;
                    img.HasAlpha = true;
                    img.VirtualPixelMethod = VirtualPixelMethod.Transparent;
                    collection.Add(img);
                 
                }

                collection.Optimize();
                collection.Write(ImagePath + "\\" + ecover_name + ".gif");

            }

**Comment by dlemstra** [comment]
Can you also share your input images?

**Reply by hafizmabuzar** [reply]
Here they are


![input_3](https://user-images.githubusercontent.com/21033535/94677857-082a3300-0337-11eb-88a9-5e2ada9ab273.png)
![input_4](https://user-images.githubusercontent.com/21033535/94677881-0d877d80-0337-11eb-8481-f0a5ad19df34.png)
![input_1](https://user-images.githubusercontent.com/21033535/94677891-11b39b00-0337-11eb-972d-7460ae6f6927.png)
![input_2](https://user-images.githubusercontent.com/21033535/94677901-15dfb880-0337-11eb-8a37-15d996525eac.png)

**Comment by hafizmabuzar** [comment]
And the actual output (previously I uploaded the smaller resized version).
![orignal_output](https://user-images.githubusercontent.com/21033535/94678324-b8983700-0337-11eb-99f6-029ad0b5c8b7.gif)

**Comment by dlemstra** [comment]
There is no solution for this. GIF only supports binary transparency. So any antialiasing would be negated by the quantization from 8-bit alpha to binary alpha.

**Reply by hafizmabuzar** [reply]
It is not possible in Magick.net or it is not achievable in any framework or image library? Can you do me a favor, please recommend any imaging library in which it is possible. Thankyou

**Reply by dlemstra** [reply]
This not possible due to the limitations of the GIF format. You will either have a transparent or an opaque pixel. Values in between are not possible in GIF and that is what will give you the smooth edges.

**Reply by hafizmabuzar** [reply]
Quite disappointed to know that. Just do me another favor by sharing the code that can put a thin feather around the edges of gif so that dithered edges may get bit blurry and non-prominent. This must be possible in magick.net. I tried to do it but could not do it. So please help. Thankyou.
